test/integration/component/maint/test_ha_pool_maintenance.py

```diff
@@ -0,0 +1,229 @@
+                    cls.clusterWithSufficientPool = cluster
+                    break
+            if not cls.clusterWithSufficientPool:
+                return
+
+            cls.services["service_offerings"][
+                "tiny"]["offerha"] = "True"
+
+            cls.services_off = ServiceOffering.create(
+                                  cls.api_client,
+                                  cls.services["service_offerings"]["tiny"])
+            cls._cleanup.append(cls.services_off)
+
+        except Exception as e:
+            cls.tearDownClass()
+            raise Exception("Warning: Exception in setup : %s" % e)
+        return
+
+    def setUp(self):
+
+        self.apiClient = self.testClient.getApiClient()
+        self.dbclient = self.testClient.getDbConnection()
+        self.cleanup = []
+        if not self.clusterWithSufficientPool:
+            self.skipTest(
+                "sufficient storage not available in any cluster for zone %s" %
+                self.zone.id)
+        self.account = Account.create(
+            self.api_client,
+            self.services["account"],
+            domainid=self.domain.id
+        )
+        self.cleanup.append(self.account)
+
+    def tearDown(self):
+        # Clean up, terminate the created resources
+        cancelmaintenance(self, storageid=self.storageid[0][0])
+        cleanup_resources(self.apiClient, self.cleanup)
+        return
+
+    @classmethod
+    def tearDownClass(cls):
+        try:
+            cleanup_resources(cls.api_client, cls._cleanup)
+        except Exception as e:
+            raise Exception("Warning: Exception during cleanup : %s" % e)
+
+        return
+
+    @attr(tags=["advanced", "cl", "advancedns", "sg",
+                "basic", "eip", "simulator", "multihost"])
+    def test_ha_with_storage_maintenance(self):
+        """put storage in maintenance mode and start ha vm and check usage"""
+        # Steps
+        # 1. Create a Compute service offering with the 'Offer HA' option
+        # selected.
+        # 2. Create a Guest VM with the compute service offering created above.
+        # 3. put PS into maintenance  mode
+        # 4. vm should go in stop state
+        # 5. start vm ,vm should come up on another storage
+        # 6. check usage events are getting generated for root disk
+
+        host = list_hosts(
+            self.api_client,
+            clusterid=self.clusterWithSufficientPool.id)
+        self.assertEqual(validateList(host)[0],
+                         PASS,
+                         "check list host response for cluster id %s"
+                         % self.clusterWithSufficientPool.id)
+
+        self.virtual_machine_with_ha = VirtualMachine.create(
+            self.api_client,
+            self.services["virtual_machine"],
+            accountid=self.account.name,
+            domainid=self.account.domainid,
+            serviceofferingid=self.services_off.id,
+            hostid=host[0].id
+        )
+
```

Comment:
I think it should be in usage test cases if not we may need to enhance migrate volume test cases.
If i add data volume , i will have to skip this test for lxc( for data volume it needs rbd storage) .which i can avoid since issue is only with root volume